cyclonedds/tools/cli/ls.py

```diff
@@ -1,27 +1,56 @@
 from threading import Thread
 import rich_click as click
-from rich import print
+from rich.console import Console
 
 from .utils import TimeDeltaParamType, LiveData, background_progress_viewer
 from .discovery.main import ls_discovery
 
 
-
 @click.command(short_help="Scan and display DDS entities in your network")
-@click.option('-i', '--id', '--domain-id', type=int, default=0, help="DDS Domain to inspect.")
-@click.option('-r', '--runtime', type=TimeDeltaParamType(), default='1s', help="Duration of discovery scan.")
-@click.option('-t', '--topic', type=str, help="Filter which entity types to display by topic name (supports regex)", default=".*")
-@click.option('--show-self', type=bool, is_flag=True, help="Show the tools own participant and subscriptions.")
-def ls(topic, id, runtime, show_self):
+@click.option(
+    "-i", "--id", "--domain-id", type=int, default=0, help="DDS Domain to inspect."
+)
+@click.option(
+    "-r",
+    "--runtime",
+    type=TimeDeltaParamType(),
+    default="1s",
+    help="Duration of discovery scan.",
+)
+@click.option(
+    "-t",
+    "--topic",
+    type=str,
+    help="Filter which entity types to display by topic name (supports regex)",
+    default=".*",
+)
+@click.option(
+    "--show-self",
+    type=bool,
+    is_flag=True,
+    help="Show the tools own participant and subscriptions.",
+)
+@click.option(
+    "--suppress-progress-bar",
+    type=bool,
+    is_flag=True,
+    help="Suppress the output of the progress bar",
+)
+@click.option(
+    "--force-color-mode",
```

Comment:
Do we also respect the `NO_COLOR` environment variable for those who don't wish to see any colors in their terminal output? See https://no-color.org/.

Reply:
Indeed, `rich` supports the `NO_COLOR` variable, which means this is not an issue purely due to the fact of choosing awesome libraries to implement things you have no clue about :innocent: